Dockerfile-python-base

```diff
@@ -0,0 +1,18 @@
+# ### Python Image ###
+FROM python:3.9.16-bullseye
+# set bash as default shell
+SHELL ["/bin/bash", "-c"]
+
+WORKDIR /app
+
+# copy requirements and setup files
+COPY requirements* ./
+COPY ape-config.yaml.example ./ape-config.yaml
+
+# install base dependencies
+RUN python -m pip install --upgrade pip
+RUN python -m pip install -r requirements.txt
+RUN ape plugins install .
+
+# install dev dependencies
+RUN python -m pip install -r requirements-dev.txt
```

Comment:
should this also install elfpy? `pip install -e .`?

Reply:
I want it to be a clean python-only image, then we install elfpy on top of it in a separate workflow.

Initially I was testing with both steps in the same flow. But I realized the "install python" step is doing the same thing every time, and taking like 3 minutes to do it. So this avoids that, by only building it when necessary (changes to requirements).

Then we only build the things that actually change (elfpy code) on top of this.